Document Nutanix (#1883)
spectrocloud/librarium: docs/docs-content/clusters/data-center/nutanix/nutanix.md

Author: Added a Getting Started

---
sidebar_label: "Nutanix"
title: "Nutanix"
description: "Learn how to configure Nutanix and create Nutanix clusters in Palette."
hide_table_of_contents: false
sidebar_position: 10
tags: ["data center", "nutanix"]
---

> [Nutanix](https://www.nutanix.com/what-we-do) is a hyper-converged infrastructure platform that combines storage, compute, and networking into a single integrated system. Palette provides a generic framework built upon the open-source [Cluster API (CAPI)](https://cluster-api.sigs.k8s.io) initiative to support adding cloud providers. As a technical preview implementing this framework, you can deploy Kubernetes clusters on Nutanix using Palette. Integration with Nutanix is achieved through a Private Cloud Gateway (PCG), which establishes a secure connection with Nutanix.

Author: It should be title case. Will make all occurrences title case and update the word list.

> The section below describes the overall workflow to register Nutanix with Palette and deploy a cluster using Palette. 

Author: applied

## Workflow

> The following summary steps outline the CAPI cloud workflow illustrated in the diagram below. A system administrator performs the first step at the system level to make the cloud provider available to all the tenants. CAPI users perform the remaining steps in Palette and their local environment.

Author: Drafted

![Diagram showing the overall CAPI workflow for users to construct self-hosted cloud solutions tailored to their environment.](/clusters_data-center_nutanix_workflow.png)

> 1. A system administrator registers Nutanix with Palette by invoking system-level APIs. This includes defining the logo, required cloud account information and key-value pairs to create the Nutanix cloud account. If you do not have a cloud logo, Palette provides a default logo. 

Reviewer: This includes defining the logo, required cloud metadata and keys required to create the Nutanix cloud account.

Palette renders a default logo if no logo is specified for any custom cloud.

Author: Replaced with suggested verbiage.

> 2. Create a self-hosted PCG by preparing YAML configuration templates and installing Palette agents for the PCG in your Kubernetes cluster using the kubectl commands provided in the Palette UI.

Reviewer: Possibly clarify the steps little bit more. Or point to the docs where self-hosted PCG is mentioned

Author: Added a link to the PCG doc.

> 2. Create a self-hosted PCG by preparing YAML configuration templates and installing Palette agents for the PCG in your Kubernetes cluster using the kubectl commands provided in the Palette UI.

Author: applied

>   When the PCG status displays Running state, the Kubernetes cluster is integrated with Palette.

Author: applied.

  :::info

>   All clouds require the deployment of a PCG, which enables Palette to securely monitor clusters in the cloud. The PCG relies on the self-hosted Kubernetes cluster to remain operational.

Author: applied

  :::

> 3. Next, add the cloud account to Palette. The required fields to add the cloud are pre-defined by the system administrator during cloud registration in step 1. The PCG name provided at the time of PCG creation is auto-filled, and you provide the account name and account details.

Reviewer: PCG name will not be auto-filled, it needs to be selected from the drop down. Can we mention select the PCG which you configured in the last step from the PCG drop down?

Author: applied

> 3. Next, add the cloud account to Palette. The required fields to add the cloud are pre-defined by the system administrator during cloud registration in step 1. The PCG name provided at the time of PCG creation is auto-filled, and you provide the account name and account details.

Author: applied

4. Create a cluster profile by selecting Nutanix as the cloud type. 

>   Palette displays out-of-the-box packs for the network and storage profile layers. Additionally, Palette provides a Nutanix-specific storage pack. If you have custom packs, you can add them to Palette by adding your registry. To learn how to add a pack registry, review [Add a Custom Registry](https://docs.spectrocloud.com/registries-and-packs/adding-a-custom-registry/#configure-a-custom-pack-registry-in-palette).

Reviewer: Palette provides a Nutanix-specific storage pack.In case you want to use OOB Nutanix CSI pack please register the cloud using the name `nutanix`

Author: Added.

5. Deploy a cluster by specifying Nutanix, listed under **Tech Preview**, as the cluster type. Select the cloud account you added and make any needed changes to the profile layers by using the YAML editor. At the **Cluster Config** step, adjust YAML configuration files to customize the cluster for your environment. You provide your own Operating System (OS) image.

> 

Reviewer: Please replace all the macros in the Cluster configuration and Node configuration pages. Macros will be mentiobed under ${}

Reviewer: you might not need to replace the ones which are already specified in the cloud account.

Author: applied

---
sidebar_label: "Nutanix"
title: "Nutanix"
description: "Learn how to configure Nutanix and create Nutanix clusters in Palette."
hide_table_of_contents: false
sidebar_position: 10
tags: ["data center", "nutanix"]
---

[Nutanix](https://www.nutanix.com/what-we-do) is a hyper-converged infrastructure platform that combines storage, compute, and networking into a single integrated system. Palette provides a generic framework built upon the open-source [Cluster API (CAPI)](https://cluster-api.sigs.k8s.io) initiative to support adding cloud providers. As a Technical Preview of implementing this framework, you can deploy Kubernetes clusters on Nutanix using Palette.  Integration with Nutanix is achieved through a Private Cloud Gateway (PCG), which establishes a secure connection with Nutanix.

## Get Started

> Once a system administrator registers Nutanix with Palette, the PCG is installed, and your Nutanix cloud account is added, you can deploy a cluster to Nutanix by using Palette. Check out the [Deploy a Cluster with Palette](../../public-cloud/deploy-k8s-cluster.md) tutorial to get started.

Reviewer 2: The semantic of the sentence is a bit confusing: I had to read it 3 times to understand that those are requirements and the PCG is not (automatically) installed after "...a system administrator registers Nutanix with Palette"

The section below describes the workflow to register Nutanix with Palette and deploy a cluster. 

> ## Workflow

Reviewer 2: Image generation is missing.

Author: Added:
"A Nutanix user with the *Prism Admin* role creates an Operating System (OS) image to be used for the Nutanix cluster that will be created in Palette. This can be done using an image creation application such as [Image Builder](https://image-builder.sigs.k8s.io/introduction.html). For guidance, review [Create Images with Image Builder](../../../byoos/image-builder.md)."

> The following summary steps outline the CAPI cloud workflow illustrated in the diagram. Since only self-hosted instances of Palette have access to the system console by a system administrator, the registration step is performed at the system level, making the cloud provider available to all the tenants. CAPI users perform the remaining steps in Palette and their local environment.

Reviewer 2: ```suggestion
The following summary steps outline the Nutanix cloud workflow illustrated in the diagram. Since only self-hosted instances of Palette have access to the system console by a system administrator, the registration step is performed at the system level, making the Nutanix cloud available to all the tenants. Regular Palette users perform the remaining steps in Palette and their local environment.
```

> ![Diagram showing the overall CAPI workflow for users to construct self-hosted cloud solutions tailored to their environment.](/clusters_data-center_nutanix_workflow.png)

Reviewer 2: ```suggestion
![Diagram showing the overall Nutanix workflow for users to construct self-hosted cloud solutions tailored to their environment.](/clusters_data-center_nutanix_workflow.png)
```

> 1. A system administrator registers Nutanix with Palette by invoking system-level APIs. This includes defining the logo, required cloud account information and key-value pairs to create the Nutanix cloud account. Palette renders a default logo if no logo is specified. 

Reviewer 2: References to the templates are missing in this bullet point.

> 2. Create a self-hosted PCG by preparing YAML configuration templates and installing Palette agents for the PCG in your Kubernetes cluster using kubectl commands provided in the Palette UI. For guidance, review [Install Private Cloud Gateway](install-pcg.md).

Reviewer 2: ```suggestion
2. Create a Nutanix self-hosted PCG by installing Palette agents in your existing Kubernetes cluster using kubectl commands provided in the Palette UI. For guidance, review [Install Private Cloud Gateway](install-pcg.md).
```

>   When the PCG status displays the state **Running**, the Kubernetes cluster is integrated with Palette.

Reviewer 2: ```suggestion
  When the PCG status displays the state **Running**, the PCG Kubernetes cluster is integrated with Palette.
```

  :::info

>   All custom providers clouds require the deployment of a PCG, which enables Palette to monitor clusters in the infrastructure provider environment. The PCG instance is installed on an existing Kubernetes cluster that must remain operational.

Reviewer 2: ```suggestion
  Nutanix cloud requires the deployment of a PCG, which enables Palette to monitor clusters in the infrastructure provider environment. The PCG instance is installed on an existing Kubernetes cluster that must remain operational.
```

  :::

> 3. Next, add the cloud account to Palette. The required fields to add the cloud are pre-defined by the system administrator during cloud registration in step 1. Use the **drop-down Menu** to select the PCG name you provided when you configured it in the previous step. You must fill out the account name and account details.

Reviewer 2: ```suggestion
3. Next, add the Nutanix cloud account to Palette. The required fields to add the cloud are pre-defined by the system administrator during cloud registration in step 1. Use the **drop-down Menu** to select the PCG name you provided when you configured it in the previous step. You must fill out the account name and account details.
```

4. Create a cluster profile by selecting Nutanix as the cloud type. 

  Palette displays out-of-the-box packs for the network and storage profile layers. Additionally, Palette provides a Nutanix-specific storage pack. If you have custom packs, you can add them to Palette by adding your registry. To learn how to add a pack registry, review [Add a Custom Registry](https://docs.spectrocloud.com/registries-and-packs/adding-a-custom-registry/#configure-a-custom-pack-registry-in-palette).

> 5. Deploy a cluster by specifying Nutanix, listed under **Tech Preview**, as the cluster type. Select the cloud account you added and make any needed changes to the profile layers by using the YAML editor. At the **Cluster Config** step, customize the Cluster configuration and Node configuration YAML files by replacing any undefined variables in curly braces, such as `${CLUSTER_NAME}`, with values for your environment. Make any other adjustments in the files to customize the cluster for your environment. You provide your own Operating System (OS) image.

Reviewer 2: You can't make any change to the OS and k8s packs.

Author: Added: "When creating a Nutanix profile, you do not have to specify anything for the OS or Kubernetes layers."

The section below describes the workflow to register Nutanix with Palette and deploy a cluster. 

## Workflow

The following summary steps outline the Nutanix cloud workflow illustrated in the diagram. Since only self-hosted instances of Palette and Palette VerteX have access to the system console by a [system administrator](../../../glossary-all.md#system-administrator), the registration step is performed at the system level, making the Nutanix cloud available to all the tenants. Regular Palette or VerteX users perform the remaining steps in Palette and their local environment.

![Diagram showing the overall Nutanix workflow for users to construct self-hosted cloud solutions tailored to their environment.](/clusters_data-center_nutanix_workflow.png)

1. A [system administrator](../../../glossary-all.md#system-administrator) registers Nutanix with Palette by preparing YAML templates to configure the cluster and node pools and invokes APIs to define the logo, required cloud account information and key-value pairs to create the Nutanix cloud account. Palette renders a default logo if no logo is specified. 

  :::caution

>   To make the Palette-provided **Nutanix CSI ---link to Nutanix CSI pack page that is being created---** pack available to users during profile creation, the Nutanix cloud must be registered using the name `nutanix`. For guidance in using registration APIs and providing required variable values, review [Register the Cloud](register-nutanix-cloud.md#register-the-cloud).

Reviewer 2: @caroldelwing I think we can remove this info box from this high level page as you have a similar one in the `Register Nutanix Cloud` page

  
  <!-- To use the **Nutanix CSI** pack, the Nutanix cloud must be registered using the name `nutanix`. For guidance in using registration APIs and providing required variable values, review [Register the Cloud](register-nutanix-cloud.md#register-the-cloud). -->
  :::

2. Create a Nutanix self-hosted PCG by installing Palette agents in your existing Kubernetes cluster using kubectl commands provided in the Palette UI. For guidance, review [Install Private Cloud Gateway](./install-pcg/install-pcg.md).

>   When the PCG status displays the state **Running**, the workload cluster is integrated with Palette.

Reviewer 2: @caroldelwing I wouldn't call the PCG a "workload cluster". Maybe something like 

```suggestion
  When the PCG status displays the state **Running**, the PCG cluster is integrated with Palette.
```

---
sidebar_label: "Nutanix"
title: "Nutanix"
description: "Learn how to configure Nutanix and create Nutanix clusters in Palette."
hide_table_of_contents: false
sidebar_position: 10
tags: ["data center", "nutanix"]
---

> [Nutanix](https://www.nutanix.com/what-we-do) is a hyper-converged infrastructure platform that combines storage, compute, and networking into a single integrated system. Palette provides a generic framework built upon the open-source [Cluster API (CAPI)](https://cluster-api.sigs.k8s.io) initiative to support adding cloud providers. As a Tech Preview of implementing this framework, you can deploy Kubernetes clusters on Nutanix using Palette. Integration with Nutanix is achieved through a Self-hosted Private Cloud Gateway (PCG), which establishes a secure connection with Nutanix Prism and allows Palette to monitor Nutanix clusters.

Author: Applied.

---
sidebar_label: "Nutanix"
title: "Nutanix"
description: "Learn how to configure Nutanix and create Nutanix clusters in Palette."
hide_table_of_contents: false
sidebar_position: 10
tags: ["data center", "nutanix"]
---

> [Nutanix](https://www.nutanix.com/what-we-do) is a hyper-converged infrastructure platform that combines storage, compute, and networking into a single integrated system. Palette provides a generic framework built upon the open-source CAPI initiative to support adding cloud providers. Using this framework, Palette supports integration with Nutanix as a *Tech Preview* for enabling cluster deployment within the Nutanix environment. This integration is achieved through a Private Cloud Gateway (PCG), which establishes a secure connection with Nutanix.

Author: Added link

The section below describes the overall workflow. 

## Workflow

The following summary steps outline the CAPI cloud workflow illustrated in the diagram below. A System Administrator performs the first step, and CAPI users perform the remaining steps in Palette.

![Diagram showing the overall CAPI workflow for users to construct self-hosted cloud solutions tailored to their environment.](/clusters_data-center_nutanix_workflow.png)

> 1. Your System Administrator registers a CAPI cloud with Palette by invoking system-level APIs. This includes defining the cloud logo, required cloud account information and key-value pairs to create the cloud account for your particular cloud. If you do not have a cloud logo, Palette provides a basic cloud logo. Palette displays the cloud when registration is complete. 

Author: Removed "CAPI" and made it just "cloud".

> <<< Does SA configure PCG in the local cluster at this step??>>>

Author: Discussed and revised sections in docs.

2. You create a self-hosted PCG for your cloud type, and install Palette agents for the PCG in your Kubernetes cluster by copying kubectl commands provided in Palette.

>   When the PCG status displays Running state, Palette integrates your Kubernetes cluster using the PCG.

Author: Revised to say: When the PCG status displays Running state, the Kubernetes cluster is now integrated with Palette.

  :::caution

>   All custom clouds require the creation of a PCG, which enables Palette to securely monitor clusters in the cloud. The PCG relies on the Kubernetes cluster in your local environment to deploy and remain operational.

Author: fixed

  :::

> 3. Next, you add the cloud account to Palette. The required fields to add the cloud are pre-defined by your System Administrator during cloud registration in step 1. These cloud-specific fields vary based on the cloud type. The PCG name provided at the time of PCG creation is auto-filled, and you provide the account name and account details.

Author: fixed

> 4. Create a cluster profile by selecting your cloud as the cloud type. You provide your own Operating System (OS) image and configure a custom Kubernetes pack layer by specifying the OS image and other settings in YAML template files during cluster creation. 

Author: Yes, moved to step 5.

>   Palette displays out-of-the-box packs for the network and storage profile layers. If you have custom packs, you can add them to Palette by adding a custom registry. To learn how to add a custom pack registry, review [Add a custom registry](https://docs.spectrocloud.com/registries-and-packs/adding-a-custom-registry/#configure-a-custom-pack-registry-in-palette).

Author: fixed

<!-- OR

Create a cluster profile by selecting your cloud as the cloud type. 

<<< Alternatively, you can provide your OS image by copying it to the provided YAML file.>>> <<<Is this correct? Will the YAML editor be blank for user to copy/paste their own image??>>> -->

> 5. Deploy a cluster by specifying the cluster type. Locate Nutanix, which is listed under **Tech Preview**. Select the cloud account you added and make any needed changes to the profile layers by using the YAML editor. At the Cluster Config step, use YAML templates to customize the cluster for your cloud environment. 

Author: added bold

<!-- Configure the control plane node and worker nodes in the provided YAML files, specifying the node pool name and the number of nodes in the pool. -->

> ## Edit and Delete Cluster

Author: Removed it

---

[Nutanix](https://www.nutanix.com/what-we-do) is a hyper-converged infrastructure platform that combines storage, compute, and networking into a single integrated system. Palette provides a generic framework built upon the open-source [Cluster API (CAPI)](https://cluster-api.sigs.k8s.io) initiative to support adding cloud providers. As a Tech Preview of implementing this framework, you can deploy Kubernetes clusters on Nutanix using Palette. Integration with Nutanix is achieved through a Self-hosted Private Cloud Gateway (PCG), which establishes a secure connection with Nutanix Prism Central and allows Palette to monitor Nutanix clusters.

## Get Started

Learn how to deploy a cluster to Nutanix by using Palette. Check out the [Deploy a Cluster with Palette](../../public-cloud/deploy-k8s-cluster.md) tutorial to get started.

The section below describes the workflow to register Nutanix with Palette and deploy a cluster. 

## Workflow

> The following summary steps outline the Nutanix cloud workflow illustrated in the diagram. Since only self-hosted instances of Palette and Palette VerteX have access to the system console by a system administrator, the registration step is performed at the system level, making the Nutanix cloud available to all the tenants. Regular Palette users perform the remaining steps in Palette and their local environment.

Reviewer 3: ```suggestion
The following summary steps outline the Nutanix cloud workflow illustrated in the diagram. Since only self-hosted instances of Palette and Palette VerteX have access to the system console by a system administrator, the registration step is performed at the system level, making the Nutanix cloud available to all the tenants. Regular Palette or VerteX users perform the remaining steps in Palette and their local environment.
```

Reviewer 4: :white_check_mark:

  
  <!-- To use the **Nutanix CSI** pack, the Nutanix cloud must be registered using the name `nutanix`. For guidance in using registration APIs and providing required variable values, review [Register the Cloud](register-nutanix-cloud.md#register-the-cloud). -->
  :::

2. Create a Nutanix self-hosted PCG by installing Palette agents in your existing Kubernetes cluster using kubectl commands provided in the Palette UI. For guidance, review [Install Private Cloud Gateway](install-pcg.md).

  When the PCG status displays the state **Running**, the workload cluster is integrated with Palette.

  :::info

  Nutanix cloud requires the deployment of a PCG, which enables Palette to monitor clusters in the infrastructure provider environment. The PCG instance is installed on an existing Kubernetes cluster that must remain operational.

  :::

3. Next, add the Nutanix cloud account to Palette. Use the **drop-down Menu** to select the PCG name you provided when you configured it in the previous step. You must fill out the account name and account details.

4. Create a cluster profile by selecting Nutanix as the cloud type. 

>   When creating a Nutanix profile, you do not have to specify anything for the OS or Kubernetes layers. Palette provides out-of-the-box packs for the network and storage profile layers, including the **Nutanix CSI** storage pack. The **Nutanix CSI** pack is available when the cloud is registered using the name `nutanix`. If you have custom packs, you can add them to Palette by adding your registry. To learn how to add a pack registry, review [Add a Custom Registry](https://docs.spectrocloud.com/registries-and-packs/adding-a-custom-registry/#configure-a-custom-pack-registry-in-palette).

Reviewer 3: This URL is incorect, use relative path instead of URL paths.

Reviewer 4: :white_check_mark:
